packages/orbit-components/src/InputGroup/index.tsx

```diff
@@ -62,12 +62,11 @@ const FakeGroup = styled.span<{
       border-radius: ${theme.orbit.borderRadiusNormal};
     `)};
 
-    &:hover {
-      box-shadow: inset 0 0 0
-        ${`${theme.orbit.borderWidthInput} ${
-          error ? theme.orbit.borderColorInputErrorHover : theme.orbit.borderColorInputHover
-        }`};
-    }
+    ${!disabled &&
```

Comment:
Instead add condition inside hover box-shadow, it won't be rendered to component if condition is false :)